sim/warlock/demonology/demonology.go

```diff
@@ -26,21 +26,62 @@ func RegisterDemonologyWarlock() {
 func NewDemonologyWarlock(character *core.Character, options *proto.Player) *DemonologyWarlock {
 	demoOptions := options.GetDemonologyWarlock().Options
 
-	demoLock := &DemonologyWarlock{
+	demonology := &DemonologyWarlock{
 		Warlock: warlock.NewWarlock(character, options, demoOptions.ClassOptions),
 	}
 
-	return demoLock
+	return demonology
 }
 
 type DemonologyWarlock struct {
 	*warlock.Warlock
 }
 
-func (demoLock *DemonologyWarlock) GetWarlock() *warlock.Warlock {
-	return demoLock.Warlock
+func (demonology DemonologyWarlock) getMasteryBonus() float64 {
+	return 0.18 + 0.023*demonology.GetMasteryPoints()
 }
 
-func (demoLock *DemonologyWarlock) Reset(sim *core.Simulation) {
-	demoLock.Warlock.Reset(sim)
+func (demonology *DemonologyWarlock) GetWarlock() *warlock.Warlock {
+	return demonology.Warlock
+}
+
+func (demonology *DemonologyWarlock) Initialize() {
+	demonology.Warlock.Initialize()
+
+	demonology.registerHandOfGuldanSpell()
+}
+
+func (demonology *DemonologyWarlock) ApplyTalents() {
+	demonology.Warlock.ApplyTalents()
+
+	// TODO: Mastery: Master Demonologist
+	// This needs to affect your own damage during metamorphosis and your pets damage at all times
+	masteryMod := demonology.AddDynamicMod(core.SpellModConfig{})
```

Comment:
make 2 mods, one for the pet and one for the warlock. Update both mods values in the mastery stat changed. Activate the pet mod in the permanent mastery auras ongain but activate the warlock mode in the meta skill. You will need to save the mod reference in demo warlock struct and access it in the meta file from there